pkg/skaffold/build/cache/lookup.go

```diff
@@ -47,8 +51,8 @@ func (c *cache) lookupArtifacts(ctx context.Context, tags tag.ImageTags, artifac
 	return details
 }
 
-func (c *cache) lookup(ctx context.Context, a *latest.Artifact, tag string) cacheDetails {
-	hash, err := c.hashForArtifact(ctx, a)
+func (c *cache) lookup(ctx context.Context, a *latest.Artifact, tag string, h artifactHasher) cacheDetails {
```

Comment:
What if you moved the artifactHasher into the `cache` instance?

Reply:
same reason as another question
>I can do this but then cache will need a Reset method that flushes the artifactHasher since I want to recompute the values on every dev loop but don't want to recompute any value within a dev loop. I think the current implementation of instantiating artifactHasher on every lookup is probably better.